File: OLD_CODES/Easy/java/TwoSum.java
package java;

class TwoSum {
    public int[] twoSum(int[] nums, int target) {

        int n = nums.length;
        int lo = 0, hi = n - 1;
        int indices[] = { 0, 1 };

        if (n == 2)
            return indices;

        while (lo < n - 1) {
            while (hi > lo) {
                if ((nums[lo] + nums[hi]) == target) {
                    indices[0] = lo;
                    indices[1] = hi;
                    return indices;
                }
                hi -= 1;
            }
            lo += 1;
            hi = n - 1;
        }
        return indices;
    }
}

File: OLD_CODES/Easy/python/twoSum.py
def twoSum(nums, target):
    n = len(nums)
    lo = 0
    hi = n - 1

    if (n == 2):
        return [0, 1]

    while (lo < n - 1) :
        while (hi > lo):
            if ((nums[lo] + nums[hi]) == target):
                return nums[lo], nums[hi]
            hi -= 1
        lo += 1
    return nums[lo], nums[hi]

print(twoSum([3,2,4], 6))